src/providers/api/Flickr.py

```diff
@@ -60,114 +60,84 @@ def getLicense(_index):
+    if raw_tag_string:
+        raw_tag_list = list(set(raw_tag_string.split()))[:max_tags]
+        return [{'name': tag.strip(), 'provider': FLICKR} for tag in raw_tag_list]
+    else:
+        return None
 
 
-    license, version = getLicense(_data.get('license'))
+def extractData(_data):
+    title = _data.get('title')
+    creator = _data.get('ownername')
+    imageURL, height, width = get_image_url(_data)
+    thumbnail = _data.get('url_s')
+    license, version = getLicense(_data.get('license', -1))
+    metaData = create_meta_data_dict(_data)
+    tagData = create_tags_list(_data)
 
-    if license is None:
-        logging.warning('License not detected!')
-        return None
+    if 'owner' in _data:
+        creatorURL = 'www.flickr.com/photos/{}'.format(_data['owner']).strip()
+    else:
+        creatorURL = None
 
-    if 'dateupload' in _data:
-        metaData['pub_date'] = sanitizeString(_data.get('dateupload'))
+    foreignID = _data.get('id')
 
-    if 'datetaken' in _data:
-        metaData['date_taken'] = sanitizeString(_data.get('datetaken'))
-
-    if 'description' in _data:
-        desc = _data.get('description', '')
-        content = None
-
-        if '_content' in desc and desc['_content'] is not None:
-            content = sanitizeString(desc.get('_content', ''))
-
-        if content:
-            metaData['description'] = content
-
-
-    if 'tags' in _data and _data['tags'] is not None:
-        tags = _data.get('tags', '').strip()
-
-        if tags:
-            maxTags = 20
-            tagData = {}
-            tagData = [{'name': tag.strip(), 'provider': 'flickr'} for tag in list(set(tags.split(' ')))[:maxTags]]
-
-
-    return [
-            imageURL if not foreignID else foreignID,
-            foreignURL,
-            imageURL,
-            thumbnail if thumbnail else '\\N',
-            str(int(float(width))) if width else '\\N',
-            str(int(float(height))) if height else '\\N',
-            '\\N',
-            license,
-            str(version) if version else '\\N',
-            creator if creator else '\\N',
-            creatorURL if creatorURL else '\\N',
-            title if title else '\\N',
-            json.dumps(metaData, ensure_ascii=False) if metaData else '\\N',
-            json.dumps(tagData, ensure_ascii=False) if tagData else '\\N',
-            'f',
-            'flickr',
-            'flickr'
-    ]
+    if foreignID and creatorURL:
+        foreignURL = '{}/{}'.format(creatorURL, foreignID)
+    else:
+        foreignURL = None
+
+
+    return create_tsv_list_row(
+        foreign_identifier=imageURL if not foreignID else foreignID,
+        foreign_landing_url=foreignURL,
+        image_url=imageURL,
+        thumbnail=thumbnail,
+        width=width,
+        height=height,
+        license=license,
+        license_version=version,
+        creator=creator,
+        creator_url=creatorURL,
+        title=title,
+        meta_data=metaData,
```

Comment:
It would be nice to change all variables to snake case but that can be part of a future refactor.

Reply:
I think I'll go ahead and do it for `Flickr.py` today.  It shouldn't take long at all.